packages/react/src/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -1,33 +1,292 @@
 import {clsx} from 'clsx'
 import type {To} from 'history'
-import React from 'react'
+import React, {useState, useRef, useCallback, useEffect, useMemo} from 'react'
 import type {SxProp} from '../sx'
 import type {ComponentProps} from '../utils/types'
 import classes from './Breadcrumbs.module.css'
 import type {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/polymorphic'
 import {BoxWithFallback} from '../internal/components/BoxWithFallback'
+import {ActionMenu} from '../ActionMenu'
+import {ActionList} from '../ActionList'
+import {IconButton} from '../Button/IconButton'
+import {KebabHorizontalIcon} from '@primer/octicons-react'
+import {useResizeObserver} from '../hooks/useResizeObserver'
+import type {ResizeObserverEntry} from '../hooks/useResizeObserver'
 
 const SELECTED_CLASS = 'selected'
 
 export type BreadcrumbsProps = React.PropsWithChildren<
   {
     className?: string
+    overflow?: 'wrap' | 'menu'
+    hideRoot?: boolean
   } & SxProp
 >
 
 const BreadcrumbsList = ({children}: React.PropsWithChildren) => {
   return <ol className={classes.BreadcrumbsList}>{children}</ol>
 }
 
-function Breadcrumbs({className, children, sx: sxProp}: BreadcrumbsProps) {
-  const wrappedChildren = React.Children.map(children, child => <li className={classes.ItemWrapper}>{child}</li>)
+type BreadcrumbsMenuItemProps = {
+  items: React.ReactElement[]
+  'aria-label'?: string
+}
+
+const BreadcrumbsMenuItem = React.forwardRef<HTMLButtonElement, BreadcrumbsMenuItemProps>(
+  ({items, 'aria-label': ariaLabel, ...rest}, ref) => {
+    return (
+      <ActionMenu>
```

Comment:
I think my personal preference would be to have this use the disclosure pattern before getting this in since using a menu for navigation is not preferred.

I assume with this work that if we went down that route we would have to do more from scratch since that's not supported in `ActionMenu` at the moment?

Reply:
Yes, We need to use a pattern involving `Details`. I tried it once on Friday and it was not straightforward. Will give it another go, now that I have a clean slate